### PR TITLE
counsel--grep-mode-occur: remove column indicators for wgrep.

### LINKED ISSUE #1800: counsel-occur broken with "ag --vimgrep "
The `ag --vimgrep` command will produce extra column in the filename line which break the wgrep edit command.

```
-*- mode:grep; default-directory: "d:/HLMJ_H5_proj/branches/HLMJ_MINA/HLMJ_creator_mina/assets/scripts/" -*-


1 candidates:
./Game/View/TableHandler/UserHandler.js:857:14:        this.CheckMJCountCorrect();
```

If I remove `--vimgrep`, the changes could be applied successfully.

I don't know whether it should be fixed in [`wgrep-emacs`](https://github.com/mhayashi1120/Emacs-wgrep) or not. I have opened an issue [here](https://github.com/mhayashi1120/Emacs-wgrep/issues/52)

PS. on Windows, if I remove the `--vimgrep` optioin, there is no preview in `counsel-ag` buffer.

### PR DESCRIPTION
**Pull request by mookid**
Fix #1800.

@zilongshanren: can you please test if this solves your issue?

### CODE REVIEW

File: counsel.el [hunk 1]
```diff
@@ -2466,7 +2466,14 @@ It applies no filtering to ivy--all-candidates."
     (insert (format "-*- mode:grep; default-directory: %S -*-\n\n\n" default-directory))
     (insert (format "%d candidates:\n" (length ivy--all-candidates)))
     (ivy--occur-insert-lines
-     (mapcar (lambda (cand) (concat "./" prepend cand)) ivy--all-candidates))))
+     (mapcar (lambda (cand)
+               (save-match-data
+                 (if (string-match
+                      ;; Remove column indicators if present
+                      "\\(?1:[^\n:]+?[^\n/:]\\):[\t ]*\\(?3:[1-9][0-9]*\\)[\t ]*:\\(?3:[1-9][0-9]*:\\)" cand)
```
Comment (basil-conto) [inline counsel.el]: Why are there three explicitly numbered groups, when only the last one is going to be used (when there are multiple capture groups with the same number, only the right-most one is used)?

File: counsel.el [hunk 2]
```diff
@@ -2466,7 +2466,14 @@ It applies no filtering to ivy--all-candidates."
...
+                     (replace-match "" nil nil cand 3)
```
Comment (basil-conto) [inline counsel.el]: The `replace-match` arguments `FIXEDCASE` and `LITERAL` should be `t`.

File: counsel.el [hunk 3]
```diff
@@ -2466,7 +2466,14 @@ It applies no filtering to ivy--all-candidates."
...
+                   cand)
```
Comment (basil-conto) [inline counsel.el]: I wonder if this can be simplified using `replace-regexp-in-string`?

File: counsel.el [hunk 4]
```diff
@@ -2466,7 +2466,14 @@ It applies no filtering to ivy--all-candidates."
...
+               (concat "./" prepend cand))) ivy--all-candidates))))
```
Comment (basil-conto) [inline counsel.el]: Nitpick: Please place `ivy--all-candidates` on a separate line, where it won't be hidden.

File: counsel.el (hunk 1 above)
Comment (mookid) [reply counsel.el]: I borrowed that from grep.el. let's rework the code after confirmation that the fix works.

**Comment by mookid** [comment]
@zilongshanren thanks for the feedback.

> I think this patch should be applied to the function `counsel-grep-like-occur` .

Probably to all 3 callers of `ivy--occur-insert-lines` indeed.

**Comment by zilongshanren** [comment]
@mookid  
I manually copy the patch code to `counsel-grep-like-occur` fun, but it is not working :(

**Comment by mookid** [comment]
I will take a look at it later today.

**Comment by mookid** [comment]
@zilongshanren could you please test again? thanks.

**Comment by zilongshanren** [comment]
@mookid Now it works, thanks. :+1:

**Comment by abo-abo** [comment]
Thanks all.

**Pull request closed without merging**